### ISSUE TITLE
Broken integration tests on Ubuntu 22.04

### ISSUE THREAD

**Issue by AndreyAkinshin** [dotnet/BenchmarkDotNet#2210]
Recently, `ubuntu-latest` GitHub action image was [switched](https://github.com/actions/runner-images/issues/6399) from 20.04 to 22.04. Now we have three permanently failing integration tests:
```
[xUnit.net]     MemoryDiagnoserSupportsNativeAOT [FAIL]
  Failed MemoryDiagnoserSupportsNativeAOT [29 s]
  Error Message:
   Assert.Equal() Failure
Expected: 32
Actual:   0
  Stack Trace:
     at BenchmarkDotNet.IntegrationTests.MemoryDiagnoserTests.AssertAllocations(IToolchain toolchain, Type benchmarkType, Dictionary`2 benchmarksAllocationsValidators) in /home/runner/work/BenchmarkDotNet/BenchmarkDotNet/tests/BenchmarkDotNet.IntegrationTests/MemoryDiagnoserTests.cs:line 278
   at BenchmarkDotNet.IntegrationTests.MemoryDiagnoserTests.MemoryDiagnoserIsAccurate(IToolchain toolchain) in /home/runner/work/BenchmarkDotNet/BenchmarkDotNet/tests/BenchmarkDotNet.IntegrationTests/MemoryDiagnoserTests.cs:line 59
   at BenchmarkDotNet.IntegrationTests.MemoryDiagnoserTests.MemoryDiagnoserSupportsNativeAOT() in /home/runner/work/BenchmarkDotNet/BenchmarkDotNet/tests/BenchmarkDotNet.IntegrationTests/MemoryDiagnoserTests.cs:line 76

[xUnit.net]     LockContentionCountIsAccurate(toolchain: ILCompiler 6.0.0-rc.1.21420.1) [FAIL]
Failed LockContentionCountIsAccurate(toolchain: ILCompiler 6.0.0-rc.1.21420.1) [17 s]
Error Message:
 System.InvalidOperationException : Sequence contains no matching element
Stack Trace:
   at System.Linq.ThrowHelper.ThrowNoMatchException()
 at BenchmarkDotNet.IntegrationTests.ThreadingDiagnoserTests.AssertStats(Summary summary, Dictionary`2 assertions) in /home/runner/work/BenchmarkDotNet/BenchmarkDotNet/tests/BenchmarkDotNet.IntegrationTests/ThreadingDiagnoserTests.cs:line 160
 at BenchmarkDotNet.IntegrationTests.ThreadingDiagnoserTests.LockContentionCountIsAccurate(IToolchain toolchain) in /home/runner/work/BenchmarkDotNet/BenchmarkDotNet/tests/BenchmarkDotNet.IntegrationTests/ThreadingDiagnoserTests.cs:line 83

[xUnit.net]     CompletedWorkItemCountIsAccurate(toolchain: ILCompiler 6.0.0-rc.1.21420.1) [FAIL]
Failed CompletedWorkItemCountIsAccurate(toolchain: ILCompiler 6.0.0-rc.1.21420.1) [17 s]
Error Message:
 System.InvalidOperationException : Sequence contains no elements
Stack Trace:
   at System.Linq.ThrowHelper.ThrowNoElementsException()
 at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
 at System.Linq.ImmutableArrayExtensions.Single[T](ImmutableArray`1 immutableArray, Func`2 predicate)
 at BenchmarkDotNet.IntegrationTests.ThreadingDiagnoserTests.AssertStats(Summary summary, Dictionary`2 assertions) in /home/runner/work/BenchmarkDotNet/BenchmarkDotNet/tests/BenchmarkDotNet.IntegrationTests/ThreadingDiagnoserTests.cs:line 156
 at BenchmarkDotNet.IntegrationTests.ThreadingDiagnoserTests.CompletedWorkItemCountIsAccurate(IToolchain toolchain) in /home/runner/work/BenchmarkDotNet/BenchmarkDotNet/tests/BenchmarkDotNet.IntegrationTests/ThreadingDiagnoserTests.cs:line 55
```

The last green CI build (Ubuntu 20.04): https://github.com/dotnet/BenchmarkDotNet/actions/runs/3514935599/jobs/5889567811
The first red CI build (Ubuntu 22.04): https://github.com/dotnet/BenchmarkDotNet/actions/runs/3534467929/jobs/5931371043

As a temporary workaround, we can rollback to 20.04 in `.github/workflows/build.yaml`, but it would be better to fix the problem.

**Comment by AndreyAkinshin** [comment]
@adamsitnik could you please take a look?

**Comment by adamsitnik** [comment]
#2209 is going to fix it (I am working on it).

@AndreyAkinshin it looks like our tests will be failing with 1h timeout on Windows again. What do you think about doing the following:

- running only .NET Core integration tests on Windows AppVeyour (it will be +- 40 minutes, there is 1h timeout)
- running all Windows tests (Full Framework + Core) on GitHub actions and Azure DevOps (no timeout)

**Comment by AndreyAkinshin** [comment]
@adamsitnik yeah, I like this plan.
P.S. I have a few ideas about how to optimize our suite of integration tests. I hope to work on this in the nearest future.

**Comment by adamsitnik** [comment]
fixed by #2209
